docs/api.yaml

```diff
@@ -0,0 +1,1061 @@
+        interventions:
+          type: array
+          items:
+            $ref: "#/components/schemas/Intervention"
+
+    Teacher:
+      type: object
+      properties:
+        id:
+          type: string
+        first_name:
+          type: string
+        last_name:
+          type: string
+        sex:
+          type: string
+          enum:
+            - M
+            - F
+        address:
+          type : string
+        phone_number:
+          type: integer
+        email:
+          type: string
+        mariage_status:
+          type: string
+        status:
+          type: string
+          description: "Defaults to ENABLED"
+          enum:
+            - ENABLED
+            - DISABLED
+        birth_date:
+          type: "string"
+          format: "date"
+        hire_date:
+          type: "string"
+          format: "date"
+        taught_courses:
+          type: array
+          items:
+            $ref: "#/components/schemas/TaughtCourse"
+        interventions:
+          type: array
+          items:
+            $ref: "#/components/schemas/Intervention"
+
+    Intervention:
+      type: object
+      properties:
+        id:
+          type: string
+        start:
+          type: "string"
+          format: "date-time"
+        end:
+          type: "string"
+          format: "date-time"
+
+    ######################################################
+    ####################### GROUP ########################
+    ######################################################
+
+    Group:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+
+    CreateGroup:
+      type: object
+      properties:
+        name:
+          type: string
+
+    ######################################################
+    ####################### COURSE #######################
+    ######################################################
+
+
+    CreateCourse:
+      type: object
+      properties:
+        name:
+          type: string
+          example: "DONNEES1"
+        total_hours:
+          type: integer
+        main_teacher_id:
+          type: string
+        main_teacher_name:
+          type: string
+
+    Course:
+      type: object
+      properties:
+        id:
+          type: string
+        name:
+          type: string
+          example: "DONNEES1"
+        total_hours:
+          type: integer
+        main_teacher_id:
+          type: string
+        main_teacher_name:
+          type: string
+
+    TaughtCourse :
+      type: object
+      properties:
+        id:
+          type: string
+        course_id:
+          type: string
+        course_name:
+          type: string
+        required_teaching_hours:
+          type: integer
+        timetable:
+          type: array
+          items:
+            $ref: "#/components/schemas/TaughtPeriod"
+
+    EnrolledCourse:
+      type: object
+      properties:
+        id:
```

Comment:
Do not put id here. Just course_id is enough. In a REST API, when you give an id, you should also give an endpoint for manipulating the identified resources. We don't need any /enrolledCourses endpoint. Similarly, also remove taughCourse.id, just use course_id

Reply:
Ok.
This conversation can be resolved, the rest is in the TaughtCourse usage's conversation.